mlflow/tracking/fluent.py

```diff
@@ -1125,7 +1128,9 @@ def search_runs(
            metrics.m tags.s.release                            run_id
         0       1.55       1.1.0-RC  5cc7feaf532f496f885ad7750809c4d4
     """
-    if not experiment_ids:
+    if search_all_experiments and (experiment_ids is None or len(experiment_ids) == 0):
+        experiment_ids = [exp.experiment_id for exp in list_experiments(view_type=ViewType.ALL)]
+    elif experiment_ids is None or len(experiment_ids) == 0:
```

Comment:
`experiment_ids is None or len(experiment_ids) == 0` can probably simplified to `not experiment_ids`

Reply:
My personal preference is to be explicit, but it looks like this change lines up with style in the rest of the repo. Will do